configs/src/networks.ts

```diff
@@ -19,17 +19,28 @@ export function getNetwork(network: SupportedNetworks): NetworkConfig | null {
   return null;
 }
 
+export function getNetworkByChainId(chainId: number): NetworkConfig | null {
+  return (
+    Object.values(networks).find(network => network.chainId === chainId) || null
+  );
+}
+
 /**
  * Retrieves the network configuration object by name or alias.
  *
- * @param {string | SupportedNetworks} network - The name or alias of the network.
+ * @param {string | SupportedNetworks | number} network - The name or alias of the network.
  * @return {NetworkConfig | null} The network configuration object if found, or `null` if not found.
  */
-export function getNetworkByNameOrAlias(
-  network: string | SupportedNetworks
+export function getNetworkByNameAliasOrChainId(
```

Comment:
move the by chain ID getter to its own function.
Functions should be small and understandable. I already pushed the limits with the get by name or alias. No need to push that even further

Reply:
having the function `getNetworkByNameOrAlias` and one `getNetworkByChainId`